### PR TITLE
Refactor Dexcom fetch runner to fix numerous issue and bugs

### PR DESCRIPTION
**Pull request by darinkrauss**
- Refactor Dexcom fetch runner to fix numerous issue and bugs
- Update Dexcom data types, parsing, and validation
- Log various Dexcom data type warnings
- Add Dexcom DataRange API and data type
- Add Dexcom Moment representing system and display time pair
- Fix Dexcom time parsing
- Add and update Dexcom-related tests
- Fix time-related prescription test bugs
- Fix time test bugs
- Update CGM Settings alert limits due to observed data
- Update Makefile version base to use constant
- Update Makefile clean targets
- Update Makefile test targets to fix compilers and use modern arguments

### CODE REVIEW

**Review by jh-bate — APPROVED** [review]
LGTM - comment about the testing constants already discussed :)

File: dexcom/alert_schedule_test.go [hunk 1]
```diff
@@ -1,226 +1,1166 @@
...
-	It("AlertSettingValueFallMgdLMinuteMinimum is expected", func() {
-		Expect(dexcom.AlertSettingValueFallMgdLMinuteMinimum).To(Equal(1.0))
+	It("AlertSettingValueHighMgdLMaximum returns expected", func() {
+		Expect(dexcom.AlertSettingValueHighMgdLMaximum).To(Equal(dataTypesSettingsCgm.HighAlertLevelMgdLMaximum))
 	})
 
-	It("AlertSettingValueHighMgdLMaximum is expected", func() {
-		Expect(dexcom.AlertSettingValueHighMgdLMaximum).To(Equal(400.0))
+	It("AlertSettingValueHighMgdLMinimum returns expected", func() {
+		Expect(dexcom.AlertSettingValueHighMgdLMinimum).To(Equal(dataTypesSettingsCgm.HighAlertLevelMgdLMinimum))
```
[the remark this answers is not in the document]
Comment (jh-bate) [reply dexcom/alert_schedule_test.go:165]: as discussed ... we found a more efficient way :)

[the remark this answers is not in the document]
Comment (darinkrauss) [reply dexcom/alert_schedule_test.go:165]: :+1:

File: data/source/source_test.go [hunk 2]
```diff
@@ -1873,6 +1873,48 @@ var _ = Describe("Source", func() {
...
+				Expect(source.HasError()).To(BeFalse())
+			})
+
+			It("returns false if the error is nil", func() {
+				source := dataSourceTest.RandomSource()
+				source.Error = &errors.Serializable{}
+				Expect(source.HasError()).To(BeFalse())
+			})
+
+			It("returns false if the error is not nil", func() {
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/source/source_test.go:1890]: Yep, typo.

[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/source/source_test.go]: Fixed and pushed.

**Review by toddkazakov — APPROVED** [review]

**Comment by darinkrauss** [comment]
Closing. Included in grouped PR https://github.com/tidepool-org/platform/pull/764.

**Pull request closed without merging**